bin/generate-nav

```diff
@@ -1,52 +1,62 @@
 #!/usr/bin/env node
 
-const fs = require('fs');
-const { markdown } = require('markdown');
-const path = require('path');
-const xpath = require('xpath');
-const yaml = require('js-yaml');
+const htmlFromMdFile = require('./htmlFromMdFile');
+const { readFileSync, writeFile } = require('fs');
+const { resolve } = require('path');
+const jsYaml = require('js-yaml');
+const loadedOutline = jsYaml.safeLoad(readFileSync(`${__dirname}/../config/outline.yaml`, 'utf8'));
+const jsyaml = require('js-yaml');
 const { DOMParser } = require('xmldom');
-
-let content = fs.readFileSync(path.resolve(__dirname, '../src/pages/docs/index.md')).toString();
-content = content.replace(/`/gm, '"');
-content = markdown.toHTML(content);
-
-const html = new DOMParser().parseFromString(content);
-
-const guessPath = h2 => xpath.select('following-sibling::ol[1]/li[1]/a/@href', h2)[0].value.split('/')[0];
-
-const normalizeAnchors = anchors =>
-  anchors.map(anchor => ({
-    id: xpath.select('./a/@href', anchor)[0].value.split('#')[1],
-    title: xpath.select('./a', anchor)[0].firstChild.data,
-  }));
-
-const normalizeItems = items =>
-  items.map((item) => {
-    const id = xpath.select('./a/@href', item)[0].value.replace('.md', '').split('/')[1];
-    const title = normalizeTitle(xpath.select('./a', item)[0].firstChild.data);
-
-    const anchors = xpath.select('./ol/li', item);
-    if (0 === anchors.length) {
-      return {
-        id,
-        title,
-      };
-    }
-
-    return {
-      id,
-      title,
-      anchors: normalizeAnchors(anchors),
-    };
+const lvl =  '  ';
+const generatedItemsId = `${lvl.repeat(2)}- id: `;
+const generatedItemsTitle = `${lvl.repeat(3)}title: `;
+const generatedAnchorId = `${lvl.repeat(4)}- id: `;
+const generatedAnchorTitle = `${lvl.repeat(5)}title: `;
+
+let yaml = '';
```

Comment:
Create an object, that you'll dump in YAML at the end using `safeDump`. The code will be easier to read, and you'll avoid a lot of issues including dealing with special characters...